Bulk upload modals

dhuangg opened the pull request.

pswk1 committed:
add Bulk Upload button to Admin Portal

add Bulk Upload button positioned on right side of Admin Portal view per Figma

pswk1 committed:
bulk upload first step

- Bulk Upload button opens first modal in flow

CalvinFronda committed:
feat(bulk): inital stepper

pswk1 committed:
style(bulk): modal content for third step

Upload Results step of bulk import flow has hard-coded content for now

pswk1 committed:
style(bulk): select display style

- style for selected template display updated to match Figma

pswk1 committed:
style(bulk): select icon

- dropdown arrow icon in select changes direction when select field is opened & closed

pswk1 committed:
style(bulk): disabled button style

refactor disabled button styling

pswk1 committed:
style(bulk): select template

bulk import default 'select template' option is hidden on dropdown

CalvinFronda committed:
feat(download): download excel

pswk1 committed:
feat(admin portal tabs): initial setup

- Use MUI Tabs for new admin portal tabs
- each tab to display a separate component

pswk1 committed:
refactor(admin portal tabs): tab view changed to object

- admin portal tab content is rendered through object properties as opposed to iterating through an array

pswk1 committed:
Apply formatting changes

pswk1 committed:
refactor(admin portal tabs): renamed settings

- renamed settings to business details in admin portal tabs

dhuangg committed:
fix(): for some reason the userType query wasn't correct

CalvinFronda committed:
feat(upload): inital upload

dhuangg committed:
feature(onboarding flow): created routing logic for stepping through each onboarding step, created onboarding control hooks to add custom submit functions, separated out onboarding controls into OnboardingControls.js, added createBusiness mutation on BizInfo form submit, finished biz form + field error handling

dhuangg committed:
feature(onboarding flow): Created business hours form, ImportFlow.js has been updated with new steps

dhuangg committed:
Merge remote-tracking branch 'origin/bulk-upload-modals' into bulk-upload-modals

CalvinFronda committed:
feat(hooks): upload & download hooks

CalvinFronda committed:
Merge branch 'bulk-upload-hooks' into bulk-upload-modals

pswk1 committed:
Merge branch 'development' into bulk-upload-modals

dhuangg committed:
feature(onboarding flow): styled form according to design spacing

dhuangg committed:
Merge branch 'development' into bulk-upload-modals

# Conflicts:
#	src/components/Authentication/LoginPage.js
#	src/components/FeatureViews/Registration/RegistrationCart/PaymentBoard.js

dhuangg committed:
feature(onboarding flow): styled business hours heading spacing

dhuangg committed:
Merge remote-tracking branch 'origin/bulk-upload-modals' into bulk-upload-modals

dhuangg committed:
fix(notes): the edit notes view doesn't have the weird formatting when opened up in "edit" mode

pswk1 committed:
Merge branch 'bulk-upload-modals' of github.com:omou-org/front-end into bulk-upload-modals

pswk1 committed:
feat(business data use case table): component

- data use case popout opens to a new tab (temporarily)

pswk1 committed:
Merge branch 'bulk-upload-modals' into admin-portal-tabs

pswk1 committed:
Apply formatting changes

CalvinFronda committed:
feat(onboard): created import templates

dhuangg committed:
Merge remote-tracking branch 'origin/bulk-upload-modals' into bulk-upload-modals

dhuangg committed:
fix(onboarding): implemented back and next logic for BulkImportStep.js, cleaned up unused files

dhuangg committed:
enhancement(onboarding): created drag and drop upload button component, styled template import results view

pswk1 committed:
feat(edit biz details): display component

- business details of Admin Portal Tabs displays biz info and an edit button

pswk1 committed:
Merge branch 'admin-portal-tabs' of github.com:omou-org/front-end into admin-portal-tabs

pswk1 committed:
refactor(business details): pull info from object

- business data info refactored to pull from object, will be pulled from a query in the future

pswk1 committed:
feat(business details): update form component

- edit button takes user to update business details form

pswk1 committed:
Apply formatting changes

CalvinFronda committed:
feat(onboarding): upload & download flows

pswk1 committed:
feat(business details): update business details

- update biz details and cache to reflect changes on UI

pswk1 committed:
Merge branch 'admin-portal-tabs' of github.com:omou-org/front-end into admin-portal-tabs

pswk1 committed:
Merge branch 'development' into admin-portal-tabs

pswk1 committed:
Merge branch 'bulk-upload-modals' into admin-portal-tabs

pswk1 committed:
fix(eslint): lint fixing

- eslint fixes

dhuangg committed:
fix(linter): cleaned up linter errors

CalvinFronda committed:
fix(lint): fixing linting

pswk1 committed:
fix(lint): lint fixes

es-lint fixes

pswk1 committed:
Merge branch 'admin-portal-tabs' of github.com:omou-org/front-end into admin-portal-tabs

dhuangg committed:
Merge pull request #771 from omou-org/admin-portal-tabs

Business Details Component

pswk1 committed:
feat(business onboarding): bizinfo form

- first step of onboarding flow has a pre-populated form of the owner's business info
- owner can make changes and/or corrections and continuing to the next step will update those changes

dhuangg committed:
enhancement(biz hours): in progress for connecting the biz hour mutation

pswk1 committed:
fix(business details): bizinfo cache update

- In onboarding flow, if user goes back a page, bizinfo is updated

pswk1 committed:
feat(business details): user use case link

- At each template import step in the onboarding flow, the "Why am I entering this data?" link will open a new tab outlining the use cases for each upload

pswk1 committed:
refactor(business details): minor refactoring

- remove unused classes and minor style fix

pswk1 committed:
Apply formatting changes

dhuangg committed:
enhancement(biz hours form): biz hours validation logic + create biz hours mutation

dhuangg committed:
enhancement(biz hours form): added closed biz days validation + submission logic. need to validate if mutation works

dhuangg committed:
fix(scheduler): fixed infinite looping of scheduler on navigation

pswk1 committed:
Merge branch 'bulk-upload-modals' into business-bulk-upload

pswk1 committed:
Apply formatting changes

pswk1 committed:
fix(biz hours form): fix from merge conflict

- update BusinessHoursForm to current code from bulk-upload-modals branch

pswk1 committed:
Merge branch 'business-bulk-upload' of github.com:omou-org/front-end into business-bulk-upload

pswk1 committed:
feat(biz hours form): update hours on onboarding

- business hours update in onboarding flow

dhuangg committed:
enhancement(biz hours form): load up biz hours data on BusinessHoursForm.js if there's existing business data

dhuangg committed:
fix(Registration): fixed useEffect dependency)

pswk1 committed:
fix(business onboarding): stepper error

- welcome step of onboarding correctly links to first step in business onboarding flow
- steps start at 1 instead of 0 for more logical UX

pswk1 committed:
Merge branch 'bulk-upload-modals' into business-bulk-upload

pswk1 committed:
Apply formatting changes

pswk1 committed:
Merge pull request #772 from omou-org/business-bulk-upload

Business bulk upload

pswk1 committed:
fix(business onboarding): stepper component

- stepper component displays correct step

CalvinFronda committed:
feat(): bulk upload, modal upload

Big boi:
Bulk upload modal
Drag and drop files
Upload files

CalvinFronda committed:
Merge branch 'bulk-upload-modals' of https://github.com/omou-org/front-end into bulk-upload-modals

CalvinFronda committed:
Merge branch 'development' into bulk-upload-modals

CalvinFronda commented on src/components/FeatureViews/Onboarding/BusinessInfo.js:
```diff
@@ -1,68 +1,222 @@
+        marginTop: '24px',
     },
     Subtitle: {
         fontFamily: 'Arial, Helvetica Neue, Helvetica, sans-serif',
```
does the fontFamily need to be here? I thought it inherits from our design system?

dhuangg replied:
yeah, we should be taking this from our font family. Why is this Arial?

dhuangg replied:
jk. this was from ryan's code... lmao. we need to fix this

CalvinFronda commented on src/components/FeatureViews/Onboarding/BusinessInfo.js:
```diff
@@ -1,68 +1,222 @@
+                        onChange={handleFieldChange(setBizName, 'name')}
+                        error={handleError(bizName, 'name')}
+                        required
+                    />{' '}
```
remove this {' '}

dhuangg replied:
you make me sad

CalvinFronda commented on src/components/Form/BusinessHoursForm.js:
```diff
@@ -0,0 +1,291 @@
+    ];
+
+    const validate = useCallback((formData) => {
+        console.log(formData);
```
remove this

CalvinFronda reviewed (commented):
TODO:
On first login, should lead to onboarding flow.

CalvinFronda commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -123,25 +143,74 @@ const BulkUploadModal = ({ closeModal }) => {
-    }
+    };
+
+    const upload = () => {
```
change to more descriptive variable name

dhuangg replied:
ditto

CalvinFronda committed:
fix(ClassEnrollmentRow): broken if statement

CalvinFronda committed:
Apply formatting changes

dhuangg commented on src/components/FeatureViews/AdminPortal/AdminPortalTabs.js:
```diff
@@ -0,0 +1,58 @@
+                onChange={handleTabChange}
+            >
+                {adminPortalTabs.map((currentTab) => (
+                    <Tab key={currentTab.tab_id} label={currentTab.label}></Tab>
```
@pswk1  Not an issue to assign the key like this, but if you just wanted to assign the key as the index of the ta you could've done this:
```
adminPortalTabs.map((currentTab, index) => (<Tab key={index} />));
```

dhuangg commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -0,0 +1,523 @@
+                                Why am I entering this data?
+                            </a>
+                            <div style={{ margin: '1em 0px' }}>
+                                <Select
```
@CalvinFronda make this Select its own Omou component since we use this dropdown throughout the design system. This component should contain the onclick logic of opening and closing/rotating the dropdown. it should also include any onClick functions passed to it when a user uses it. 

Example:
```
<OmouSelect onClick={someFunction}/>
```
I should be able to see the arrow on the right side toggle up and down as well as see `someFunction()` get executed.

dhuangg commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -0,0 +1,523 @@
+                                data you want to upload, fill out the template
+                                and upload it back Omou.
+                            </Typography>
+                            <a
```
@pswk1 use the [Navlink react router component](https://reactrouter.com/web/api/NavLink) because the routing may get confused by "href"

dhuangg commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -0,0 +1,523 @@
+        switch (step) {
+            case 0:
+                return (
+                    <Grid container className={classes.modalStyle}>
```
@CalvinFronda  / @pswk1 Separate each step into its own component

dhuangg commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -0,0 +1,523 @@
+            case 1:
+                return (
+                    <>
+                        {loading ? (
```
Why is the loading logic handled here instead of at the component level before the return statement?

dhuangg replied:
adjust the loading component to handle this case
```
<Loading isLoading={loading}>
    <UploadTemplateStep/>
</Loading>
```

dhuangg commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -0,0 +1,523 @@
+                                                    <button
+                                                        onClick={removeUpload}
+                                                    >
+                                                        X
```
@CalvinFronda use the [close icon](https://material-ui.com/components/material-icons/) from material ui and replace `<button/>` with `<IconButton/>` from material ui

dhuangg commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -0,0 +1,523 @@
+                                            onChange={handleSelectFile}
+                                        />
+                                        {fileName ? (
+                                            <>
```
since the `<p>` tag is 1 tag you won't need a template literal to wrap everything

dhuangg commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -0,0 +1,523 @@
+                                variant='body1'
+                            >
+                                {`${
+                                    uploadResponse?.data[`upload${template}`]
```
@CalvinFronda extract to an explanatory variable

dhuangg commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -0,0 +1,523 @@
+                                        .totalFailure
+                                } rows failed.`}
+                            </Typography>
+                            {uploadResponse?.data[`upload${template}`]
```
ditto, but also shouldn't you have this load if uploadResponse is invalid?

dhuangg commented on src/components/FeatureViews/AdminPortal/BusinessDetails.js:
```diff
@@ -0,0 +1,452 @@
+        switch (step) {
+            case 0:
+                return (
+                    <Grid
```
@pswk1 separate these out into 2 different components (one ViewBusinessDetails and EditBusinessDetails)

dhuangg commented on src/components/FeatureViews/Onboarding/BusinessHours.js:
```diff
@@ -0,0 +1,55 @@
+
+const BusinessHours = () => {
+    const classes = useStyles();
+    // const handleChange = (newValue, actionMeta) => {};
```
delete comment

dhuangg commented on src/components/FeatureViews/Onboarding/DragAndDropUploadBtn.js:
```diff
@@ -0,0 +1,101 @@
+    const style = useMemo(
+        () => ({
+            ...baseStyle,
+            ...(isDragActive ? activeStyle : {}),
```
@CalvinFronda we can make a small helper function for this: `applyStyle(isDragActive, activeStyle)`

CalvinFronda replied:
This is boiler plate from the library we are using: https://react-dropzone.js.org/#section-styling-dropzone

dhuangg commented on src/components/FeatureViews/Onboarding/DragAndDropUploadBtn.js:
```diff
@@ -0,0 +1,101 @@
+        () => ({
+            ...baseStyle,
+            ...(isDragActive ? activeStyle : {}),
+            ...(isDragAccept ? acceptStyle : {}),
```
replace all instances of this with the applyStyle() helper function

dhuangg commented on src/components/FeatureViews/Onboarding/ImportResults.js:
```diff
@@ -0,0 +1,122 @@
+                        <br />
+                        <Typography align='left'>
+                            {`${
+                                state.UPLOAD_RESPONSE.data[
```
separate this out into an explanatory variable

dhuangg commented on src/components/FeatureViews/Onboarding/ImportResults.js:
```diff
@@ -0,0 +1,122 @@
+                        <br />
+                        <Typography align='left'>
+                            {`${
+                                state.UPLOAD_RESPONSE.data[
```
ditto

dhuangg reviewed (commented):
some cleanup, but the logic is overall solid

CalvinFronda committed:
fix(): bulk upload cleanup

All comments https://github.com/omou-org/front-end/pull/774#pullrequestreview-661629424

CalvinFronda committed:
chore(bulkUpload): separate modal steps into own components

CalvinFronda commented:
> TODO:
> On first login, should lead to onboarding flow.

Still needs to be done before merging.

CalvinFronda committed:
Apply formatting changes

CalvinFronda commented on src/components/OmouComponents/OmouSelect.js:
```diff
@@ -0,0 +1,94 @@
+
+export default OmouSelect;
+
+OmouSelect.propTypes = {
```
Ill make these required

CalvinFronda commented on src/components/FeatureViews/AdminPortal/BulkUploadModal.js:
```diff
@@ -106,10 +108,10 @@ const GET_TEMPLATE = {
 
 const BulkUploadModal = ({ closeModal }) => {
-    const [template, setTemplate] = useState('');
+    const [selectedItem, setSelectedItem] = useState('');
```
I feel like `selectedTemplate` would be a better variable name for this. lmk what you think

dhuangg commented on src/components/FeatureViews/AdminPortal/AdminPortalTabs.js:
```diff
@@ -0,0 +1,58 @@
+    const classes = useStyles();
+
+    const adminPortalTabs = [
+        { label: 'overview', tab_id: 0 },
```
isn't the id just the index? and i don't think we're using the tab_index so may want to consider deleting?

dhuangg reviewed (commented):
yolo.

CalvinFronda committed:
Merge branch 'development' into bulk-upload-modals

CalvinFronda committed:
fix(): loading handled in UploadTemplateStep

fix comments

CalvinFronda committed:
Merge branch 'bulk-upload-modals' of https://github.com/omou-org/front-end into bulk-upload-modals

CalvinFronda committed:
Apply formatting changes

CalvinFronda committed:
Revert "Merge branch 'bulk-upload-modals' of https://github.com/omou-org/front-end into bulk-upload-modals"

This reverts commit e418c8cbcd7dc598f84b5098eeb7fe09a0891be9, reversing
changes made to 73a7393c26aadd2b2f63d202cded88e213781d10.

CalvinFronda committed:
Merge branch 'bulk-upload-modals' of https://github.com/omou-org/front-end into bulk-upload-modals

CalvinFronda commented:
when i merge i get ```getGoogleClassroomCourses``` is no a function. Check in development but it looks like it doesn't exist. Is this a Michael issue? 

@dhuangg

dhuangg commented:
rip. probably something went wrong with a merge. @andrewjknapp @ml5885 any insight on this one?

CalvinFronda committed:
chore(omouselect): required propTypes

dhuangg committed:
Merge branch 'development' into bulk-upload-modals

# Conflicts:
#	src/components/Authentication/LoginPage.js
